### ISSUE TITLE
Subscription not starting after page routing in `NextJS`

### ISSUE THREAD

**Issue by 413n** [supabase/realtime#254]
# Bug report

## Describe the bug

I have a `NextJS` app with 2 pages using the same hook that sets up the subscription to a table.
The hook is pretty simple with a useEffect that looks like this (it's basically copied from the [nextjs-slack-clone hook](https://github.com/supabase/supabase/blob/master/examples/slack-clone/nextjs-slack-clone/lib/Store.js)):
```javascript
  // ...
  useEffect(() => {
    const messageListener = sb
      .from<Message>("message")
      .on("INSERT", (payload) => setTempNewMessage(payload.new))
      .subscribe((event, e) => {
        console.warn(`[Message] Listener ${event}`, messageListener);
      });

    console.error("STARTED LISTENING...", messageListener);
    
    // Cleanup on unmount
    return () => {
      messageListener
        .unsubscribe()
        .receive("ok", () => console.error("!SUBSCRIPTION REMOVED!"));
    };
  }, []);
```

At the `first page` load, the subscription is all good and works perfectly.
But when I do `router.push` to the `other page`, it `unsubscribe` correctly but the new subscription starts `CLOSED` and even the `RETRYING_AFTER_TIMEOUT` does nothing.

The execution seems in sequence as I can see from the console (so the problem should not be there):

![image](https://user-images.githubusercontent.com/16388025/166085642-0435d075-7097-405a-9c17-089d6b27101a.png)

Checking on the `websocket` connection I saw these messages:
- `1st and 2nd`: First page load
- `3rd`: `unsubscribe` inside the `useEffect unmount` of the first page
- `4th and 5th`: `join` from the new page (should be good)
- `6th`: closes the subscription for no reason (**Here is the bug**)

![image](https://user-images.githubusercontent.com/16388025/166085665-e7a2d5fb-5e8e-48fb-bd92-937c36ae0ac8.png)

@garyaustin1 helped me notice that the timespan between the `leave` and the `join` (in the second page) was really really short (`0.008s`), even tho they are still sequentially.

So then I tried putting a delay of `1 second` at page load before starting the `subscription` and it **worked**!

As @garyaustin1 suggested it could be a server glitch caused by the fast `leave` and `join`.

## To Reproduce

`NextJS` project with 2 different pages (the `useEffect` must `unmount`) that starts a `subscription` at page load.

## Expected behavior

The subscription should stay `SUBSCRIBED` after second page load.

## System information

- OS: Windows
- Browser: Chrome
- Version of @supabase/supabase-auth-helpers: 1.3.0
- Version of supabase-js: 1.30.0
- Version of Node.js: 16.13.0

## Additional context

I'm using `@supabase/supabase-auth-helpers`. The version of the supabase-js then it's a bit behind the latest version (at time of writing `1.35.2`).

You can the see the Discord discussion from the thread started on this message: https://discord.com/channels/839993398554656828/843999948717555735/969764357342920745

**Comment by GaryAustin1** [comment]
I was relooking at your code and wonder if the problem is not awaiting the disconnect to finish.    You might try calling the supabase.js removeSubsciption method also. https://supabase.com/docs/reference/javascript/removesubscription

**Comment by 413n** [comment]
@GaryAustin1 I tried also using `supabase.removeSubscription(messageListener)` but it does not work. It's even more delayed because that action is async and you can't execute `await` inside the unmount in React (as long as I know).

**Comment by w3b6x9** [comment]
@413n can you call `sb.getSubscriptions()` and see what the output is?

another thing you can do is call and await `sb.removeAllSubscriptions()` before you subscribe to any subscriptions.

**Comment by 413n** [comment]
> @413n can you call `sb.getSubscriptions()` and see what the output is?

I did not re-tried but as far as I remember, debugging `sb.getSubscriptions()` after the subscription creation in the second page, it was an empty array (`[]`).

> another thing you can do is call and await `sb.removeAllSubscriptions()` before you subscribe to any subscriptions.

I tried also `sb.removeAllSubscriptions()` but it was not working neither. I couldn't really use it tho because I had another `app`-level subscription. 


I don't have much more insights because for the use case I needed, in the end I put the subscription at the `app`-level and I'm not encountering this problem anymore.
